src/Storages/System/StorageSystemDDLWorkerQueue.cpp

```diff
@@ -0,0 +1,289 @@
+                res_columns[i++]->insert(address.host_name); // host_name
+                auto resolved = address.getResolvedAddress();
+                res_columns[i++]->insert(resolved ? resolved->host().toString() : String()); // host_address
+                res_columns[i++]->insert(address.port); // port
+                ddl_query_path = fs::path(ddl_zookeeper_path) / queries[query_id];
+
+                zkutil::Strings active_nodes;
+                zkutil::Strings finished_nodes;
+
+                // on error just skip and continue.
+
+                code = zookeeper->tryGetChildren(fs::path(ddl_query_path) / "active", active_nodes);
+                if (code != Coordination::Error::ZOK && code != Coordination::Error::ZNONODE)
+                    zk_exception_code = code;
+
+                code = zookeeper->tryGetChildren(fs::path(ddl_query_path) / "finished", finished_nodes);
+                if (code != Coordination::Error::ZOK && code != Coordination::Error::ZNONODE)
+                    zk_exception_code = code;
+
+                /* status:
+                 * active: If the hostname:port entry is present under active path.
+                 * finished: If the hostname:port entry is present under the finished path.
+                 * errored: If the hostname:port entry is present under the finished path but the error count is not 0.
+                 * unknown: If the above cases don't hold true, then status is unknown.
+                 */
+                if (std::find(active_nodes.begin(), active_nodes.end(), address.toString()) != active_nodes.end())
+                {
+                    res_columns[i++]->insert(static_cast<Int8>(Status::active));
+                }
+                else if (std::find(finished_nodes.begin(), finished_nodes.end(), address.toString()) != finished_nodes.end())
+                {
+                    if (pool_status[replica_index].error_count != 0)
+                    {
+                        res_columns[i++]->insert(static_cast<Int8>(Status::errored));
+                    }
+                    else
+                    {
+                        res_columns[i++]->insert(static_cast<Int8>(Status::finished));
+                    }
+                    // regardless of the status finished or errored, the node host_name:port entry was found under the /finished path
+                    // & should be able to get the contents of the znode at /finished path.
+                    auto res_fn = zookeeper->asyncTryGet(fs::path(ddl_query_path) / "finished");
+                    auto stat_fn = res_fn.get().stat;
+                    query_finish_time = stat_fn.mtime;
+                }
+                else
+                {
+                    res_columns[i++]->insert(static_cast<Int8>(Status::unknown));
+                }
+
+                //  This is the original cluster_name from the query (WHERE cluster='cluster_name')
+                res_columns[i++]->insert(cluster_name);
```

Comment:
We should obtain `cluster_name` from DDL query (the query that is written into `/clickhouse/task_queue/ddl/query-0000000000` node), not from `SELECT ... FROM system.distributed_ddl_queue`

Reply:
`cluster_name` now extracted from DDL query (referred from DDLWorker.cpp)